egs/csj/s5/local/nnet/run_cnn2d.sh

```diff
@@ -0,0 +1,136 @@
+  steps/compute_cmvn_stats.sh $train $train/log $train/data || exit 1;
+  # Split the training set
+  utils/subset_data_dir_tr_cv.sh --cv-spk-percent 10 $train ${train}_tr90 ${train}_cv10
+fi
+
+# Run the CNN pre-training,
+if [ $stage -le 1 ]; then
+  dir=exp/cnn2d4
+  ali=${gmm}_ali_nodup
+  # Train
+  $cuda_cmd $dir/log/train_nnet.log \
+    steps/nnet/train.sh \
+      --cmvn-opts "--norm-means=true --norm-vars=true" \
+      --delta-opts "--delta-order=2" --splice 5 \
+      --network-type cnn2d --cnn-proto-opts "--pool1-y-len=4 --pool1-y-step=4 --pitch-dim=3" \
+      --hid-layers 2 --learn-rate 0.008 \
+      ${train}_tr90 ${train}_cv10 data/lang $ali $ali $dir || exit 1;
+  # Decode
+  for eval_num in eval1 eval2 eval3 $dev_set ;do
+      steps/nnet/decode.sh --nj 10 --cmd "$decode_cmd" --config conf/decode_dnn.config --acwt 0.2 \
+    $gmm/graph_csj_tg data-fbank/$eval_num $dir/decode_${eval_num}_csj || exit 1;
+  done
+fi
+
+# Pre-train stack of RBMs on top of the convolutional layers (4 layers, 1024 units),
+if [ $stage -le 2 ]; then
+  dir=exp/cnn2d4_pretrain-dbn
+  transf_cnn=exp/cnn2d4/final.feature_transform_cnn # transform with convolutional layers
+  # Train
+  $cuda_cmd $dir/log/pretrain_dbn.log \
+    steps/nnet/pretrain_dbn.sh --nn-depth 4 --hid-dim 1024 --rbm-iter 20 \
+    --feature-transform $transf_cnn --input-vis-type bern \
+    --param-stddev-first 0.05 --param-stddev 0.05 \
+    $train $dir || exit 1
+fi
+
+# Re-align using CNN,
+if [ $stage -le 3 ]; then
+  dir=exp/cnn2d4
+  steps/nnet/align.sh --nj 10 --cmd "$train_cmd" \
+    $train data/lang $dir ${dir}_ali || exit 1
+fi
+
+# Train the DNN optimizing cross-entropy,
+if [ $stage -le 4 ]; then
+  dir=exp/cnn2d4_pretrain-dbn_dnn; [ ! -d $dir ] && mkdir -p $dir/log;
+  ali=exp/cnn2d4_ali
+  feature_transform=exp/cnn2d4/final.feature_transform
+  feature_transform_dbn=exp/cnn2d4_pretrain-dbn/final.feature_transform
+  dbn=exp/cnn2d4_pretrain-dbn/4.dbn
+  cnn_dbn=$dir/cnn_dbn.nnet
+  { # Concatenate CNN layers and DBN,
+    num_components=$(nnet-info $feature_transform | grep -m1 num-components | awk '{print $2;}')
+    cnn="nnet-copy --remove-first-layers=$num_components $feature_transform_dbn - |"
+    nnet-concat "$cnn" $dbn $cnn_dbn 2>$dir/log/concat_cnn_dbn.log || exit 1 
+  }
+  # Train
+  $cuda_cmd $dir/log/train_nnet.log \
+    steps/nnet/train.sh --feature-transform $feature_transform --dbn $cnn_dbn --hid-layers 0 \
+    ${train}_tr90 ${train}_cv10 data/lang $ali $ali $dir || exit 1;
+  # Decode (reuse HCLG graph)
+  for eval_num in eval1 eval2 eval3 $dev_set ;do
+      steps/nnet/decode.sh --nj 10 --cmd "$decode_cmd" --config conf/decode_dnn.config --acwt 0.2 \
+    $gmm/graph_csj_tg data-fbank/$eval_num $dir/decode_${eval_num}_csj || exit 1;
+  done
+fi
+
+# Sequence training using sMBR criterion, we do Stochastic-GD 
+# with per-utterance updates. For RM good acwt is 0.2,
+dir=exp/cnn2d4_pretrain-dbn_dnn_smbr
+srcdir=exp/cnn2d4_pretrain-dbn_dnn
+acwt=0.2
```

Comment:
For 'normal' databases, the sMBR is done with 'acwt=0.1'. Only RM was an exception, because it is small and has non-standard LM... Please take the DNN setup from 'wsj' or 'swbd'.